src/structures/CommandInteraction.js

```diff
@@ -0,0 +1,222 @@
+'use strict';
+
+const APIMessage = require('./APIMessage');
+const Interaction = require('./Interaction');
+const WebhookClient = require('../client/WebhookClient');
+const { Error } = require('../errors');
+const { ApplicationCommandOptionTypes, InteractionResponseTypes } = require('../util/Constants');
+const MessageFlags = require('../util/MessageFlags');
+
+/**
+ * Represents a command interaction.
+ * @extends {Interaction}
+ */
+class CommandInteraction extends Interaction {
+  constructor(client, data) {
+    super(client, data);
+
+    /**
+     * The ID of the invoked application command
+     * @type {Snowflake}
+     */
+    this.commandID = data.data.id;
+
+    /**
+     * The name of the invoked application command
+     * @type {string}
+     */
+    this.commandName = data.data.name;
+
+    /**
+     * Whether the reply to this interaction has been deferred
+     * @type {boolean}
+     */
+    this.deferred = false;
+
+    /**
+     * The options passed to the command.
+     * @type {CommandInteractionOption[]}
+     */
+    this.options = data.data.options?.map(o => this.transformOption(o, data.data.resolved)) ?? [];
+
+    /**
+     * Whether this interaction has already been replied to
+     * @type {boolean}
+     */
+    this.replied = false;
+
+    /**
+     * An associated webhook client, can be used to create deferred replies
+     * @type {WebhookClient}
+     */
+    this.webhook = new WebhookClient(this.applicationID, this.token, this.client.options);
+  }
+
+  /**
+   * The invoked application command, if it was fetched before
+   * @type {?ApplicationCommand}
+   */
+  get command() {
+    const id = this.commandID;
+    return this.guild?.commands.cache.get(id) ?? this.client.application.commands.cache.get(id) ?? null;
+  }
+
+  /**
+   * Defers the reply to this interaction.
+   * @param {boolean} [ephemeral] Whether the reply should be ephemeral
+   * @returns {Promise<void>}
+   * @example
+   * // Defer the reply to this interaction
+   * interaction.defer()
+   *   .then(console.log)
+   *   .catch(console.error)
+   * @example
+   * // Defer to send an ephemeral reply later
+   * interaction.defer(true)
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  async defer(ephemeral) {
+    if (this.deferred || this.replied) throw new Error('INTERACTION_ALREADY_REPLIED');
+    await this.client.api.interactions(this.id, this.token).callback.post({
+      data: {
+        type: InteractionResponseTypes.DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE,
+        data: {
+          flags: ephemeral ? MessageFlags.FLAGS.EPHEMERAL : undefined,
+        },
+      },
+    });
+    this.deferred = true;
+  }
+
+  /**
+   * Options for a reply to an interaction.
+   * @property {boolean} [ephemeral] Whether the reply should be ephemeral
+   * @typedef {WebhookMessageOptions} InteractionReplyOptions
```

Comment:
```suggestion
   * Options for a reply to an interaction.
   * @typedef {WebhookMessageOptions} InteractionReplyOptions
   * @property {boolean} [ephemeral] Whether the reply should be ephemeral
```
Shouldn't the typedef be above the property?